lang/cs/Org.Apache.REEF.IO.Tests/TestHadoopFileSystem.cs

```diff
@@ -128,24 +129,44 @@ public void TestGetChildren()
             _fileSystem.DeleteDirectory(remoteDirectory);
         }
 
-        [Fact(Skip = "These tests need to be run in an environment with HDFS installed.")]
+        [Fact(Skip = SkipMessage)]
         public void TestOpen()
         {
             // Open() is not supported by HadoopFileSystem. Use CopyToLocal and open the local file instead.
             Assert.Throws<NotImplementedException>(() => _fileSystem.Open(GetTempUri()));
         }
 
-        [Fact(Skip = "These tests need to be run in an environment with HDFS installed.")]
+        [Fact(Skip = SkipMessage)]
         public void TestCreate()
         {
             // Create() is not supported by HadoopFileSystem. Create a local file and use CopyFromLocal instead.
             Assert.Throws<NotImplementedException>(() => _fileSystem.Create(GetTempUri()));
         }
 
+        /// <summary>
+        /// Tests whether .IsDirectory() works.
+        /// </summary>
+        [Fact(Skip = SkipMessage)]
+        public void TestIsDirectory()
+        {
+            var remoteDirUri = GetTempUri();
+            _fileSystem.CreateDirectory(remoteDirUri);
+            var remoteFakeuri = GetTempUri();
+            var remoteFileUri = GetTempUri();
+            var localFile = FileSystemTestUtilities.MakeLocalTempFile();
+            _fileSystem.CopyFromLocal(localFile, remoteFileUri);
+            Assert.True(_fileSystem.IsDirectory(remoteDirUri));
+            Assert.False(_fileSystem.IsDirectory(remoteFakeuri));
+            Assert.False(_fileSystem.IsDirectory(remoteFileUri));
+            _fileSystem.DeleteDirectory(remoteDirUri);
+            _fileSystem.Delete(remoteFileUri);
+            File.Delete(localFile);
```

Comment:
This cleanup should be handled in the test cleanup.

Reply:
Existing tests follow this same pattern

---
In reply to: [194556624](https://github.com/apache/reef/pull/1473#discussion_r194556624) [](ancestors = 194556624)

Reply:
Agreed, but please avoid in the future if possible. Please break up this code block for readability as well.

---
In reply to: [194580410](https://github.com/apache/reef/pull/1473#discussion_r194580410) [](ancestors = 194580410,194556624)

Reply:
fixed

---
In reply to: [195218460](https://github.com/apache/reef/pull/1473#discussion_r195218460) [](ancestors = 195218460,194580410,194556624)